packages/ember-data/lib/system/store.js

```diff
@@ -250,7 +252,8 @@ Store = Ember.Object.extend({
 
     if (DS.Adapter.detect(adapter)) {
```

Comment:
This seems inconsistent to me. I mean, I thought we now should always lookup adapter/serializer via the container and not instanciating it. Also, we encourage strings everywhere, but in this case we still want to fully support a store definition with an adapter property a class ?